subprojects/language-native/src/main/java/org/gradle/language/swift/tasks/UnexportMainSymbol.java

```diff
@@ -82,40 +86,58 @@ public DirectoryProperty getOutputDirectory() {
     }
 
     @TaskAction
-    public void unexport() {
-        for (final File file: source) {
-            final File relocatedObject = outputDirectory.file(file.getName()).get().getAsFile();
-            if (OperatingSystem.current().isWindows()) {       
-                try {
-                    final SymbolHider symbolHider = new SymbolHider(file);
-                    symbolHider.hideSymbol("main");     // 64 bit
-                    symbolHider.hideSymbol("_main");    // 32 bit
-                    symbolHider.saveTo(relocatedObject);
-                } catch (IOException e) {
-                    throw UncheckedException.throwAsUncheckedException(e);
-                }   
+    public void unexport(InputChanges inputChanges) {
```

Comment:
❓ Shouldn't the `getObjects()` property be annotated with `@Incremental`?

Reply:
`SkipWhenEmpty` implies `@Incremental`. The task would fail when querying a non-incremental input.